code/modules/projectiles/gun_helpers.dm

```diff
@@ -536,6 +536,22 @@ DEFINES in setup.dm, referenced here.
 
 	var/obj/item/active_hand = get_active_hand()
 	if(active_hand)
+		if(active_hand.preferred_storage)
+			for(var/storage in active_hand.preferred_storage)
+				var/list/items_in_slot
+				if(islist(get_item_by_slot(active_hand.preferred_storage[storage])))
+					items_in_slot = get_item_by_slot(active_hand.preferred_storage[storage])
+				else
+					items_in_slot = list(get_item_by_slot(active_hand.preferred_storage[storage]))
+				
+				for(var/item_in_slot in items_in_slot)
+					if(istype(item_in_slot, storage))
+						var/slot = active_hand.preferred_storage[storage]
+						if(slot == WEAR_ACCESSORY)
+							slot = WEAR_IN_ACCESSORY
```

Comment:
You should implement the rest of the WEAR_IN kinds else `preferred_storage` is kinda pointless for anything but accessories (For example `WEAR_WAIST` would need to be `WEAR_IN_BELT`